aptos-move/move-examples/nft_auction_house/sources/auction_house_own_listing.move

```diff
@@ -0,0 +1,498 @@
+    }
+
+    public entry fun initialize_auction_house(
+        account: &signer,
+        market_fee_numerator: u64,
+        market_fee_denominator: u64,
+        fee_address: address,
+    ) {
+        move_to(
+            account,
+            AuctionHouseConfig {
+                market_fee_denominator,
+                market_fee_numerator,
+                fee_address,
+            }
+        );
+    }
+
+    public entry fun initialize_auction<CoinType>(account: &signer) {
+        move_to(
+            account,
+            Auctions<CoinType> {
+                cur_auction_id: 0,
+                all_active_auctions: table::new(),
+                listing_event: account::new_event_handle<ListingEvent>(account),
+                bid_event: account::new_event_handle<BidEvent>(account),
+                cancel_bid_events: account::new_event_handle<CancelBidEvent>(account),
+
+            }
+        );
+    }
+
+    public fun generate_auction_data<CoinType>(
+        owner: &signer,
+        creator: address,
+        collection_name: String,
+        token_name: String,
+        property_version: u64,
+        amount: u64,
+        min_price: u64,
+        start_sec: u64, // specify when the auction starts
+        expiration_sec: u64, // specify when the auction ends
+        min_incremental_price: u64,
+    ): Auction<CoinType> {
+        //
+        let sec = timestamp::now_seconds();
+        assert!(sec <= start_sec, error::invalid_argument(EINVALIDE_START_TIME));
+        assert!(start_sec < expiration_sec, error::invalid_argument(EINVALIDE_EXPIRATION_TIME));
+        let token_id = token::create_token_id_raw(creator, collection_name, token_name, property_version);
+        let listing = listing::create_list<CoinType>(
+            owner,
+            token_id,
+            amount,
+            min_price,
+            false,
+            start_sec,
+            expiration_sec
+        );
+
+        Auction<CoinType>{
+            listing,
+            bids: vector::empty(),
+            highest_price: 0,
+            highest_bid: guid::create_id(@auction_house, 0),
+            min_incremental_price,
+        }
+    }
+
+    public entry fun create_auction<CoinType>(
+        owner: &signer,
+        creator: address,
+        collection_name: String,
+        token_name: String,
+        property_version: u64,
+        amount: u64,
+        min_price: u64,
+        start_sec: u64, // specify when the auction starts
+        expiration_sec: u64, // specify when the auction ends
+        min_incremental_price: u64,
+    ) acquires Auctions {
+        create_auction_no_entry<CoinType>(
+            owner,
+            creator,
+            collection_name,
+            token_name,
+            property_version,
+            amount,
+            min_price,
+            start_sec,
+            expiration_sec,
+            min_incremental_price,
+        );
+    }
+
+    public fun create_auction_no_entry<CoinType>(
```

Comment:
we need to figure out naming here and stick to a convention :P

Reply:
updated